### PR TITLE
Report pipeline failures to slack

### PR DESCRIPTION
**Pull request by dwaynebailey**
Third time lucky using direct calls to the notofication functions.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -6,9 +6,16 @@ import uk.gov.hmcts.contino.Environment
...
-  sharedInfrastructurePipeline('ccd', environment.prodName, 'prod')
+    try {
+        sharedInfrastructurePipeline('ccd', environment.nonProdName, 'nonprod')
+        sharedInfrastructurePipeline('ccd', environment.demoName, 'nonprod')
+        sharedInfrastructurePipeline('ccd', environment.prodName, 'prod')
+    } catch (err) {
+          notifyBuildFailure channel: slackChannel
+          throw err
+    }
+    notifyBuildFixed channel: slackChannel
```
Comment (vlaurin) [inline Jenkinsfile_CNP]: wouldn't it make more sense to have `notifyBuildFixed` as the last line inside the `try` block? that way you don't rely on the `catch` block rethrowing the error.

Comment (dwaynebailey) [reply Jenkinsfile_CNP]: I mostly mapped it to how that had been implemented, but does make it more readable where you say. I just re-throw'd cause I guessed Jenkins wanted to know.  I'll keep the throw and move the Fixed call.

Comment (dwaynebailey) [reply Jenkinsfile_CNP]: OK just did what you said :) Its more readable now, tx.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -6,10 +6,16 @@ import uk.gov.hmcts.contino.Environment
...
-  }
-  sharedInfrastructurePipeline('ccd', environment.demoName, 'nonprod')
-  sharedInfrastructurePipeline('ccd', environment.prodName, 'prod')
+    try {
+        sharedInfrastructurePipeline('ccd', environment.nonProdName, 'nonprod')
+        sharedInfrastructurePipeline('ccd', environment.demoName, 'nonprod')
+        sharedInfrastructurePipeline('ccd', environment.prodName, 'prod')
+        notifyBuildFixed channel: slackChannel
+    } catch (err) {
+          notifyBuildFailure channel: slackChannel
```
Comment (vlaurin) [inline Jenkinsfile_CNP]: indentation is a bit off

**Pull request merged**